### ISSUE TITLE
Be more flexible on path value: must be absolute if not Windows Store target

### ISSUE THREAD

**Issue by ianwieds** [Teamwork/node-auto-launch#105]
- The target platform(s) the problem occurs on (Linux, Mac, Mac App Store, and or Windows): Windows Store
- Node version (run `node -v`): v12.18.3
- auto-launch version: 5.0.5
- The options you're passing to auto-launch: name, path
- Are you using NW.js or Electron?
  - If so, which version?: 11.1.1 (latest as of now)
  - If you're using Electron, are you using auto-launch from the main process or a renderer process? main
- Can you reproduce this with a simple Hello World example app? yes
- If needs be, would you be able to provide us with a simple app (as simple as possible) which reproduces the problem?

---

We are trying out a test to see how this works for Windows Store but we're getting an error: `path must be absolute`

```
  const autoLaunch = new AutoLaunch({
    name: 'OurApp',
    path: 'explorer.exe shell:AppsFolder\\OurCompany.OurApp_abcd123456!OurCompany.OurApp',
  });

// The error throws immediately: 'path must be absolute'
```

We followed the instructions in the docs. Any help would be appreciated!

**Comment by jurkog** [comment]
Running in to the same issue, any update on this?

**Comment by racoonman** [comment]
This issue is still occurring, any update?

**Comment by miguelx97** [comment]
Hello!
I've just made a new PR solving this problem with Microsoft store appx: https://github.com/Teamwork/node-auto-launch/pull/111
Example of how you can instance `autoLaunch`: `new autoLaunch({ name: 'APP_NAME', path:"DEV_ID.APP_ID!PACKAGE_NAME", isHidden: true })`

I hope you find it useful :)

**Comment by serkanyazicioglu** [comment]
I guess @Teamwork members don't look at here anymore 😢

**Comment by hovancik** [comment]
Please merge the fix :)

**Comment by Oxalin** [comment]
The next version bump will include support for Windows AppX, the code has been merged and reworked.

**Comment by Oxalin** [comment]
Thanks to @miguelx97 for the PR. I based the feature on your code with some tweaks. This should know be available from 5.1.0.

**Comment by Oxalin** [comment]
Please, use 5.1.0 for this feature.

**Comment by hovancik** [comment]
@Oxalin npm installs 5.0.6 , will you please update? https://www.npmjs.com/package/auto-launch

**Comment by Oxalin** [comment]
Once the source code is updated, I have no clue on how to update the
information on npmjs. If someone can point me in the right direction, it
will be a pleasure to update it.

I also have a new major version on my boiling plate which will get rid of
CoffeeScript, have better code coverage and add new functionalities. I'm
almost ready to merge it and it could be added as a beta version on npmjs
for test purposes in the same way.

**Comment by Oxalin** [comment]
This may answer your request, I'll have a look this weekend:
https://docs.npmjs.com/updating-your-published-package-version-number

**Comment by Oxalin** [comment]
@4ver @adam-lynch Any possibility for one of you to add either add me under npmjs for the project or to launch a "npm publish" on your side?

**Comment by hovancik** [comment]
@Oxalin I am trying to install from github with `npm i Teamwork/node-auto-launch` but getting error when trying to use it:

```
Error: Cannot find module 'C:\Users\conta\stretchly\node_modules\auto-launch\dist\index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:445:19)
    at Module._findPath (node:internal/modules/cjs/loader:687:18)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1070:27)
    at n._resolveFilename (node:electron/js2c/browser_init:2:114728)
    at Module._load (node:internal/modules/cjs/loader:929:27)
    at l._load (node:electron/js2c/asar_bundle:2:13642)
    at Module.require (node:internal/modules/cjs/loader:1150:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at App.initialize (C:\Users\conta\stretchly\app\main.js:226:20)
    at App.emit (node:events:529:35)
 ```

**Comment by Oxalin** [comment]
I'll fix it. I didn't know that was a prerequiste.

**Comment by 4ver** [comment]
It's not a prerequisite/we don't want to push the js 👌

**Comment by 4ver** [comment]
I just setup automatic publishes to npm when releases are published on github.

To release, create a tag and release with the same name (semver version number) as seen here:

<img width="989" alt="Screenshot 2024-02-27 at 15 59 06" src="https://github.com/Teamwork/node-auto-launch/assets/1408434/ab14f3cb-2493-4a93-ad90-fb83011f587d">

You can release to a beta channel on npm by adding `-rcX` to the end of the release version. If you don't add `-rcX` **it will release to production**. Please take care when making releases that you use the correct version numbers.

Ping me if you want to release to production.

**Comment by 4ver** [comment]
Also fyi, i released version 6.0.0-rc1 with the changes from master https://github.com/Teamwork/node-auto-launch/releases/tag/6.0.0-rc1

**Comment by Oxalin** [comment]
Thank you! I'm short on time right now and I haven't had the time to deal my own not-committed changes.
